Lib/test/test_imaplib.py

```diff
@@ -440,6 +440,19 @@ def test_simple_with_statement(self):
         with self.imap_class(*server.server_address):
             pass
 
+    def test_imaplib_timeout_test(self):
+        _, server = self._setup(SimpleIMAPHandler)
+        addr = server.server_address[1]
+        client = self.imap_class("localhost", addr, timeout=10.0)
+        self.assertEqual(client.sock.timeout, 10.0)
+        client.shutdown()
+        client = self.imap_class("localhost", addr, timeout=None)
+        self.assertEqual(client.sock.timeout, None)
+        client.shutdown()
+        client = self.imap_class("localhost", addr, timeout=socket._GLOBAL_DEFAULT_TIMEOUT)
+        self.assertEqual(client.sock.timeout, None)
```

Comment:
I wouldn't expect this to pass. Shouldn't it be `self.assertEqual(client.sock.timeout, socket._GLOBAL_DEFAULT_TIMEOUT)`?

Reply:
@ericvsmith 
This is due to socket module implementation.
https://github.com/python/cpython/blob/d565fb9828ee9c494bb7a80057a08e4738273e30/Modules/socketmodule.c#L6677

Reply:
https://github.com/python/cpython/blob/d565fb9828ee9c494bb7a80057a08e4738273e30/Modules/socketmodule.c#L1002